src/OpenTelemetry.Api/Context/BaggageContext.cs

```diff
@@ -0,0 +1,241 @@
+﻿// <copyright file="BaggageContext.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenTelemetry.Context
+{
+    /// <summary>
+    /// Baggage context.
+    /// </summary>
+    public readonly struct BaggageContext : IEquatable<BaggageContext>
```

Comment:
I think we should name it just `Baggage`. 
https://w3c.github.io/baggage/

Reply:
I have a different opinion on this. This is similar to the fact that we have SpanContext while W3C has TraceContext.
The W3C spec is describing the wire format. The in process representation and class name don't have to (or even should not) follow a name mentioned in the wire format spec.

Reply:
We can call it whatever OpenTelemetry spec decide to call it. It looked to me like the plan is to replace CorrelationContext to Baggage.
https://github.com/open-telemetry/opentelemetry-specification/issues/536
If its called BaggageContext, then we are good here.

Reply:
Yes, totally agree. We should follow the OpenTelemetry spec (unless the name is already taken by .NET - such like Span 😆).

Reply:
When I was doing this work, I had it called "Baggage" initially. But for some reason, I felt like the spec would stick with keeping "Context" in the name and it would end up being "BaggageContext" instead. It's really just a guess, we can always rename when it settles?

Reply:
Yes, lets keep BaggageContext for now.

Reply:
Changed name to `Baggage`